pkg/controller/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -970,6 +973,28 @@ func (r *ReconcileAWSPrivateLink) reconcileHostedZone(awsClient *awsClient,
 	return modified, hostedZoneID, nil
 }
 
+// selectDomain determines if the specified domain is a subdomain of any item
+// in the list and returns the most accurately matching one. Default to the
+// subdomain when no matches are found.
+func selectDomain(subdomain string, domains []string) string {
+	// The subdomain is empty, no results found.
+	if subdomain == "" {
+		return subdomain
+	}
+	bestMatch := ""
+	for _, domain := range domains {
```

Comment:
I have to verify this in go playground every time: no nil check necessary here ✓